### PR TITLE
[Merged by Bors] - feat(RingTheory/AlgebraicIndependent): add some results on `AlgebraicIndependent`

### PR DESCRIPTION
**Pull request by acmepjz**
- `algebraicIndependent_unique_type_iff`, `algebraicIndependent_iff_transcendental`: equivalency of `AlgebraicIndependent` and `Transcendental` for one-element family
- `AlgebraicIndependent.transcendental`: f a family `x` is algebraically independent, then any of its element is transcendental
- `MvPolynomial.(algebraicIndependent|transcendental)_X`: algebraic independent properties of intermediates of `MvPolynomial`
- `algebraicIndependent_of_finite'`: variant of `algebraicIndependent_of_finite` using `Transcendental`
- `exists_isTranscendenceBasis'`: `Type` version of `exists_isTranscendenceBasis`

Also changed one `¬IsAlgebraic` to `Transcendental`.

Also add `Set.subtypeInsertEquivOption` (used in the proof of `algebraicIndependent_of_finite'`) which is similar to `Finset.subtypeInsertEquivOption`. Note that it does not have simp lemmas yet, as the existing `Finset` one has no simp lemmas, either.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

This PR towards the result that if `E/F` is transcendental, then there are infinitely many `F`-embeddings from `E` to its algebraic closure.

### CODE REVIEW

File: Mathlib/RingTheory/AlgebraicIndependent.lean [hunk 1]
```diff
@@ -82,6 +83,21 @@ theorem algebraicIndependent_empty_type_iff [IsEmpty ι] :
     AlgebraicIndependent R x ↔ Injective (algebraMap R A) := by
   rw [algebraicIndependent_iff_injective_aeval, MvPolynomial.aeval_injective_iff_of_isEmpty]
 
+/-- An element `x` is transcendental if and only if the one-element family `![x]` is
+algebraically independent. -/
+theorem transcendental_iff_algebraicIndependent {x : A} :
```
Comment (erdOne) [inline Mathlib/RingTheory/AlgebraicIndependent.lean:88]: I think it's useful to have a `AlgebraicIndependent R x ↔ Transcendental R (x default)` given `Unique ι` as a simp lemma.

Comment (acmepjz) [reply Mathlib/RingTheory/AlgebraicIndependent.lean:88]: Good idea. Will be added soon.

**Comment by erdOne** [comment]
Thanks!
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

File: Mathlib/RingTheory/AlgebraicIndependent.lean [hunk 2]
```diff
@@ -432,6 +484,13 @@ theorem exists_isTranscendenceBasis (h : Injective (algebraMap R A)) :
   simp only [Subtype.range_coe_subtype, setOf_mem_eq] at *
   exact hs.2.eq_of_le ⟨ht, subset_univ _⟩ hst
 
+/-- `Type` version of `exists_isTranscendenceBasis`. -/
+theorem exists_isTranscendenceBasis' (R : Type u) {A : Type v} [CommRing R] [CommRing A]
+    [Algebra R A] (h : Injective (algebraMap R A)) :
+    ∃ (ι : Type v) (x : ι → A), IsTranscendenceBasis R x := by
```
[the remark this answers is not in the document]
Comment (acmepjz) [reply Mathlib/RingTheory/AlgebraicIndependent.lean:490]: Ah, personally I prefer a `Type` version... and indeed it is used in another of my PR...

[the remark this answers is not in the document]
Comment (alreadydone) [reply Mathlib/RingTheory/AlgebraicIndependent.lean:490]: If you really want to get rid of the coercion you can write `set ι : Type _ := s` in your proof ... (not sure it works, but you get the idea)

[the remark this answers is not in the document]
Comment (acmepjz) [reply Mathlib/RingTheory/AlgebraicIndependent.lean:490]: > If you really want to get rid of the coercion you can write `set ι : Type _ := s` in your proof (not sure it works, but you get the idea) ... maybe `set x := Subtype.val` too

Then I need `clear_value` to hide the actual definition of `ι`. I think `exists_isTranscendenceBasis'` vs `exists_isTranscendenceBasis` is like [Module.Free.exists_basis](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/FreeModule/Basic.html#Module.Free.exists_basis) vs [Module.free_iff_set](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/FreeModule/Basic.html#Module.free_iff_set), it's convenient to have both.

[the remark this answers is not in the document]
Comment (alreadydone) [reply Mathlib/RingTheory/AlgebraicIndependent.lean:490]: I'd say the definition of Module.Free is nonstandard. It shows your lemma is not without precedents though, so I guess it's fine to keep it.

**Pull request closed without merging**